src/templates/TemplateData.ts

```diff
@@ -133,107 +133,35 @@ function verifyTemplatesByRuntime(templatesMap: { [runtime: string]: Template[]
     }
 }
 
-export async function tryGetTemplateDataFromCache(reporter: TelemetryReporter | undefined, globalState: vscode.Memento, cliFeedJson: cliFeedJsonResponse): Promise<TemplateData | undefined> {
-    try {
-        return <TemplateData | undefined>await callWithTelemetryAndErrorHandling('azureFunctions.tryGetTemplateDataFromCache', reporter, undefined, async function (this: IActionContext): Promise<TemplateData | undefined> {
-            this.suppressErrorDisplay = true;
-            this.properties.isActivationEvent = 'true';
-            const templatesMap: { [runtime: string]: Template[] } = {};
-            const configMap: { [runtime: string]: Config } = {};
-            for (const key of Object.keys(ProjectRuntime)) {
-                // called within loop in case setting has changed between runtimes
-                const userTemplateVersion: string | undefined = getFuncExtensionSetting(templateVersionSetting);
-                const runtime: ProjectRuntime = <ProjectRuntime>ProjectRuntime[key];
-                const feedRuntime: string = getFeedRuntime(runtime);
-                const releaseVersion: string = userTemplateVersion ? userTemplateVersion : cliFeedJson.tags[feedRuntime].release;
-                const cachedRelease: string | undefined = globalState.get(`${runtime}-release`);
-                if (!cachedRelease || releaseVersion !== cachedRelease) {
-                    // templates are not up-to-date and need to be downloaded/extracted
-                    return undefined;
-                }
-
-                const cachedResources: object | undefined = globalState.get<object>(getRuntimeKey(resourcesKey, runtime));
-                const cachedTemplates: object[] | undefined = globalState.get<object[]>(getRuntimeKey(templatesKey, runtime));
-                const cachedConfig: object | undefined = globalState.get<object>(getRuntimeKey(configKey, runtime));
-
-                if (cachedResources && cachedTemplates && cachedConfig) {
-                    [templatesMap[runtime], configMap[runtime]] = parseTemplates(cachedResources, cachedTemplates, cachedConfig);
-                    verifyTemplatesByRuntime(templatesMap, runtime);
-                } else {
-                    return undefined;
-                }
-            }
-
-            return new TemplateData(templatesMap, configMap);
-        });
-    } catch (error) {
-        return undefined;
-    }
-}
-
-export async function tryGetLatestTemplateData(reporter: TelemetryReporter | undefined, cliFeedJson: cliFeedJsonResponse, globalState?: vscode.Memento): Promise<TemplateData | undefined> {
+export async function tryGetTemplateData(reporter: TelemetryReporter | undefined, cliFeedJson: cliFeedJsonResponse, globalState?: vscode.Memento, isBackup: boolean = false): Promise<TemplateData | undefined> {
+    const templateReleaseVersion: string = 'templateReleaseVersion';
+    const v1ReleaseVersion: string = '1.0.12'; // known stable version
```

Comment:
I think these should go in 'constants.ts'. They're pretty important and I think will be easier to find that way.